common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/AuthorizationActivity.java

```diff
@@ -23,19 +23,29 @@
 package com.microsoft.identity.common.internal.providers.oauth2;
 
 import android.os.Bundle;
+import android.util.Log;
+import android.webkit.ClientCertRequest;
+import android.widget.Toast;
 
 import androidx.annotation.Nullable;
 import androidx.fragment.app.Fragment;
 
 import com.microsoft.identity.common.internal.ui.DualScreenActivity;
+import com.microsoft.identity.common.java.challengehandlers.IChallengeHandler;
 import com.microsoft.identity.common.logging.Logger;
 
+import java.util.concurrent.locks.ReentrantLock;
+
 public class AuthorizationActivity extends DualScreenActivity {
 
     public static final String TAG = AuthorizationActivity.class.getSimpleName();
 
     private AuthorizationFragment mFragment;
 
+    //Holds ChallengeHander that handles CBA.
+    public static IChallengeHandler<ClientCertRequest, Void> sClientCertAuthChallengeHandler;
```

Comment:
I'm wondering if we can do something like this

https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1728
(The code above will definitely break. I just dished it out as a guideline.)

Reply:
(I also think that we're handling state bundle incorrectly all along).

Reply:
I assume making the class serializable is not an option. If it is, then that is always the best way to go.